### ISSUE TITLE
Additional argument to InstanceD

### ISSUE THREAD

**Issue by Icelandjack** [ekmett/lens#649]
There is an additional argument to `InstanceD` ([migration guide](https://ghc.haskell.org/trac/ghc/wiki/Migration/8.0?action=diff&version=42)) so lens fails to compile with GHC head (8.1.20160419):

```haskell
data Dec = ... | InstanceD (Maybe Overlap) Cxt Type [Dec] | ...
```

Just needs to change [Language.Haskell.TH.Lens](https://github.com/ekmett/lens/blob/bdb7e9142e83ff92a3f914a7ffc7ea4f5582a6d9/src/Language/Haskell/TH/Lens.hs#L851):

```haskell
_InstanceD :: Prism' Dec (Maybe Overlap, Cxt, Type, [Dec])
_InstanceD
  = prism' reviewer remitter
  where
      reviewer (x, y, z, ö) = InstanceD x y z ö
      remitter (InstanceD x y z ö) = Just (x, y, z, ö)
      remitter _ = Nothing
```

and `Control.Lens.TH`:

```haskell
 InstanceD overlap ctx inst body -> InstanceD overlap ctx inst <$> traverse go body
```

I'm not sure about the version bounds

**Comment by RyanGlScott** [comment]
This will be fixed once #648 is merged. At this point, we're just waiting for GHC 8.0.1 in [hvr's Ubuntu PPA](https://launchpad.net/~hvr/+archive/ubuntu/ghc) to update (which will happen when 8.0-rc4 is released).

**Comment by Icelandjack** [comment]
Great

**Comment by phadej** [comment]
ping @RyanGlScott, hvr's ppa is updated.

**Comment by RyanGlScott** [comment]
Fixed in 0e83bd6eea46ee66b53be0e7ffcc9256299c1e69.
